hpy/devel/include/common/macros.h

```diff
@@ -22,15 +26,64 @@
      ...macros.h:28:19: error: invalid type argument of unary ‘*’ (...)
      (void)sizeof((*data)->_ob_head._reserved0), / * ERROR: blah blah * /
 
+   Unfortunately, HPy_New now supports non-legacy types which do not contain
+   PyObject_HEAD, so this check is no longer correct in all cases.
 */
 
 #define HPy_New(ctx, cls, data) (_HPy_New(                                    \
     (ctx),                                                                    \
     (cls),                                                                    \
-    ((void)sizeof((*data)->_ob_head._reserved0), /* ERROR: expected a variable of type T** with T a structure starting with HPyObject_HEAD */ \
-     (void**)data)                                                            \
+    ((void**)data)                                                            \
   ))
 
+/* A macro for creating (static inline) helper functions for custom types.
+
+   Two versions of the helper exist. One for legacy types and one for pure
+   HPy types.
+
+   Example for a pure HPy custom type:
+
+       HPyType_HELPERS(PointObject)
+
+   This would generate the following:
+
+   * `PointObject * PointObject_AsStruct(HPyContext ctx, HPy h)`: a static
+     inline function that uses HPy_AsStruct to return the PointObject struct
+     associated with a given handle. The behaviour is undefined if `h`
+     is associated with an object that is not an instance of PointObject.
+
+   * `PointObject_STRUCT_IS_LEGACY`: an enum value set to 0 so that in the
+     HPyType_Spec for PointObject one can write
+     `.legacy = PointObject_STRUCT_IS_LEGACY` and not have to remember to
+     update the spec when the helpers used changes.
+
```

Comment:
what about just `PointObject_IS_LEGACY`? And possibily rename `.legacy` to `.is_legacy`, so what we can write the following?
```c
HPyType_Spec = {
    ...
    .is_legacy = PointObject_IS_LEGACY,
    ...
}
```

Reply:
Happy to remove the `STRUCT_` part. I started switching to `is_legacy` but it feels too different to any of the other field names.

Reply:
Renaming to remove `STRUCT_` done.